js/views/conversation_view.js

```diff
@@ -620,7 +627,10 @@
         messageDetail: function(e, data) {
             var view = new Whisper.MessageDetailView({
                 model: data.message,
-                conversation: this.model
+                conversation: this.model,
+                // we pass these in to allow nested panels
+                listenBack: this.listenBack.bind(this),
+                resetPanel: this.resetPanel.bind(this)
```

Comment:
I'm still not crazy about these triple(+?)ly nested panels. It makes it a little harder to keep track of the fact that you're not just making a decision about a certain contact, but a certain message. But then again, hopefully our new key change policies ensure that users reach this UX even more rarely than they used to.